lib/charms/mongodb/v1/shards_interface.py

```diff
@@ -231,10 +239,25 @@ def add_shards(self, departed_shard_id):
                     logger.info("Adding shard: %s ", shard)
                     mongo.add_shard(shard, shard_hosts)
                 except PyMongoError as e:
+                    # raise exception after trying to add the remaining shards, as to not prevent
+                    # adding other shards
                     logger.error("Failed to add shard %s to the config server, error=%r", shard, e)
-                    raise
+                    failed_to_add_shard = (e, shard)
+
+        if not failed_to_add_shard:
+            self.charm.unit.status = ActiveStatus("")
```

Comment:
maybe we can move this 
```
self.charm.unit.status = ActiveStatus("")
return
```
under this 
```
mongo.add_shard(shard, shard_hosts)
```
in that case we won't need if and we can move the processing of error code under `except`.

What do you think?